### PR TITLE
doc: Glossary of jargonish terms for layman in the context of Devtron

### PR DESCRIPTION
**Pull request by ashokdevtron**

### CODE REVIEW

File: docs/reference/glossary.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Glossary
+
+* **Artifacts**
+
+    Any files or resources generated during the software development lifecycle, including source code, binary files, and compiled packages. In Devtron, you can view the artifacts in the `Build History` tab of your application.
+
+* **Base Deployment Template**
+
+    It’s a single entry point (an all-in-one YAML file) for you to enter the values, so that when the application is deployed the filled values go to the respective template files (YAML), and accordingly the resources would be created. In Devtron, you get the option to select a base deployment template in the `App Configuration` tab at the time of creating an application.
```
Comment (abhibhaw) [inline docs/reference/glossary.md:9]: Na this description too is not appropriate

Comment (abhibhaw) [reply docs/reference/glossary.md:9]: And also we need to define Deployment Template before base Deployment Template?

File: docs/reference/glossary.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+* **Build Context**
+
+    The set of files and directories used as input when building a Docker container image using a Dockerfile. Specify the set of files to be built by referring to a specific subdirectory, relative to the root of your repository. To build all files from the root, use (.) as the build context, or set build context by referring a subdirectory path such as /myfolder or /myfolder/buildhere if path not set, default path will be root dir of selected git repository.
```
Comment (abhibhaw) [inline docs/reference/glossary.md:13]: This too is incorrect we don't need a set of files as an input. Let's say for building a docker image: we require a dockerfile and context. Dockerfile contains the instructions to build. Context is the path where the build process may refer for files required for build

**Review by abhibhaw — CHANGES_REQUESTED** [review]

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: docs/reference/glossary.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+In Devtron, you can choose a deployment strategy by going to Applications (choose your app) → App Configuration (tab) → Workflow Editor → (edit deployment pipeline) → Deployment Strategy. [Read More...](https://docs.devtron.ai/usage/applications/creating-application/cd-pipeline#deployment-strategies)
+
+### Devtron Agent
+
+Your Kubernetes cluster gets mapped with Devtron when you save the cluster configurations. Now, the Devtron agent (rollout controller) must be installed from the chart store on the added cluster so that you can deploy your applications on that cluster. [Read More...](https://docs.devtron.ai/global-configurations/cluster-and-environments#installing-devtron-agent)
+
+### Devtron Apps
+
+The containerization of your source code followed by its deployment is totally configurable if you create a Devtron App. Custom apps deployed on Kubernetes cluster using the CI/CD feature of Devtron are called Devtron Apps. To create one, go to Applications → Create (button) → Custom App.
```
Comment (Abhinav-26) [inline docs/reference/glossary.md:85]: This definition is good but looks a bit confusion?? Can we have something like,

Devtron Apps are the micro-services deployed with Kubernetes-native CI/CD with Devtron.

**Review by Abhinav-26 — APPROVED** [review]
Looks good to me.

**Review by abhibhaw — APPROVED** [review]
LGTM. Thanks @ashokdevtron 🎉 🎉 Approving this but please don't merge I guess devops team has some ongoing release in progress.

**Pull request merged**